client/landing/stepper/declarative-flow/internals/steps-repository/ecommerce-segmentation-survey/mutations/use-save-answers-mutation.ts

```diff
@@ -0,0 +1,35 @@
+import { useMutation } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+type SaveAnswersHookArgs = {
+	surveyKey: string;
+	userId?: number | null;
+	anonId?: string;
+	blogId?: number;
+};
+
+type SaveAnswersMutationArgs = {
+	questionKey: string;
+	answerKeys: string[];
+};
+
+const useSaveAnswersMutation = ( { surveyKey, userId, anonId, blogId }: SaveAnswersHookArgs ) => {
+	return useMutation( {
+		mutationFn: async ( { answerKeys, questionKey }: SaveAnswersMutationArgs ) => {
+			return wpcom.req.post( {
+				path: `/segmentation-survey`,
+				apiNamespace: 'wpcom/v2',
+				body: {
+					user_id: userId,
+					anon_id: anonId,
```

Comment:
Based on your feedback, the endpoint gets user_id and anon_id from cookies, not from the request body.

Reply:
Indeed, I'll update that (: